openedx/frontend-app-course-authoring
# feat: break up load api to tab specific

JIRA Ticket: [TNL-11244](https://2u-internal.atlassian.net/browse/TNL-11244)
> Users are seeing the page load. The load time is very long and after the loading spinner goes away the page does not load correctly. The API is throwing a 504 (Gateway timeout) error.

This PR is dependent on `edx-platform` PR #[33909](https://github.com/openedx/edx-platform/pull/33909)

This PR separates the api call for the home page into three parts:
1. Basic functionality call
2. Courses data call
3. Libraries data call

Separating the one api call into these three helps spread out the work on the api and help prevent timeout errors. The PR also adds error messages after each of the api calls so users will know when the api call has failed. Separating out the api will also help developers more easily find and resolve bugs in the loading process.

Testing
1. Open the Network tab in DevTools
5. Navigate to the Studio home page
6. Page should load as expected tabs and number of courses
7. Look at the Network tab and confirm that the `home` api call did not include courses, archived courses, or libraries
8. Look at the Network tab and confirm that the `courses` api call was successful
9. Click on the Library tab
10. Should show spinner while loading libraries
11. After loading completes, expected libraries should be listed
12. In the `home` api call, update the url to be invalid
13. Reload the page
14. Page should load with failure alert
15. Check Network tab and confirm the `courses` api was not called
16. Fix the `home` api url and update the `courses` api to be invalid
17. Reload the page
18. The page should load the tabs and add course/library buttons
19. The course tab should show failure alert
20. Click the library tab
21. Should load libraries as expected
22. Fix the `courses` api and update the `libraries` api to be invalid
23. Reload the page
24. The page should load as expected with all the courses
25. Click the library tab
26. Tab should show failure alert
